apps/test/unit/code-studio/utilsTest.js

```diff
@@ -34,15 +34,23 @@ describe('utils', () => {
 
   describe('updateQueryParam', () => {
     let pushStateOrig = window.history.pushState;
+    let replaceStateOrig = window.history.replaceState;
```

Comment:
In sinon, every stub is also a spy (although every spy is not a stub).

> Test stubs are functions (spies) with pre-programmed behavior.
> 
> They support the full test spy API in addition to methods which can be used to alter the stub’s behavior.
>
> ([from the top of this doc page](http://sinonjs.org/releases/v2.4.1/stubs/))

I think it would work like this:
```js
beforeEach(() => {
  sinon.stub(window.history, 'pushState');
  sinon.stub(window.history, 'replaceState');
});

afterEach(() => {
  window.history.pushState.restore();
  window.history.replaceState.restore();
});

it('can use replaceState', () => {
  // ...
  updateQueryParam('param1', undefined, true);
  assert(window.history.pushState.notCalled);
  assert(window.history.replaceState.calledOnce);
  assert.equal(window.history.replaceState.firstCall.args[2], '/some/path?param2=two');
});
```